README.md

```diff
@@ -14,16 +14,21 @@ You can use this action in an existing workflow and have it run after a linter o
 
 ```yaml
 name: 'markdownlint'
+
 on:
   pull_request:
     paths: ['**/*.md']
 
+permissions:
+  contents: read
+  pull-requests: write
+
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
```

Comment:
You could even do better to show the most best-practice pattern (I couldn't find the right word here), and explicitly clear all permissions at the workflow level, with

```yaml
permissions: {}
```

And then, at the job level, to add only the required scopes, with what you have now.

Apart from being clear that you completely choose what permissions are used, it also comes into play when adding new jobs to the same workflow in the future, where you wouldn't mistakenly let them run with more permissions than you'd expect.

Reply:
Fair point! I probably wouldn't clear all permissions at the workflow level, because [default permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) for workflows are typically managed in repository (or organization) settings, so this could create a troubleshooting headache. However, it would be safer to move the permissions configuration to the job where we need these permissions, like so:

```suggestion
jobs:
  lint:
    runs-on: ubuntu-latest
    permissions:
      contents: read
      pull-requests: write
    steps:
```

Pull requests welcome! 😉